src/components/ClaimForm.js

```diff
@@ -175,6 +180,9 @@ class ClaimForm extends Component {
     if (this.state.claim.dateClaimed < this.state.claim.dateFrom) return false;
     if (!!this.state.claim.dateTo && this.state.claim.dateFrom > this.state.claim.dateTo) return false;
     if (!this.state.claim.icd) return false;
+    if (this.isExplanationMandatoryForIPD){
+      if (this.state.claim.visitTypeOutInPatient=="IPD" && !this.state.claim.explanation) return false;
```

Comment:
Substitute '===' in place of '=='. Additionally, you might contemplate introducing a constant for "IPD" using a more descriptive name. As of now, the meaning of "IPD" is not entirely apparent to me.